src/util-unittest.h

```diff
@@ -119,6 +119,17 @@ extern int unittests_fatal;
         PASS;              \
     } while (0)
 
+/**
+ * \brief Pass the test if expression evaluates to true.
+ *
+ * Only to be used at the end of a function instead of returning the
```

Comment:
Hmm but we do not seem to require to use it at the end of the function in detect-mark.c, right?

Reply:
I added that since the PASS_IF macro also had that description in util-unittest.h, though it wasn't required to use at the end of function in detect-mark.c. Please do correct me if I am wrong.

Reply:
hmm I see that `PASS_IF` is only used at one place and it really is used at the end of the function. Although there could be places when it is required in the middle of a function. This makes me wonder if PASS_* statements are always expected be at the end of the function (which means detect-mark test should be modified in that manner) or should this comment be updated. @victorjulien ?

Reply:
Should i update the comment and make the new pull request?

Reply:
hmm I'd wait for Victor to respond. @victorjulien ?

Reply:
@victorjulien, please advice on whether to update the comment.

Reply:
The goal of new (and updated) unittests should be that they run from start to finish (top to bottom) in the pass case. So a pass mid-function violates that. We don't want any logic that runs in a 'fail' case, other than checking for the fail itself. No cleanup, no additional checks in the failure path.

Reply:
So, a pass function must ultimately run after ensuring that there are no failures, right? So, shall i add a comment as below @victorjulien 

> Only to be used to check pass case after checking all fail cases

Reply:
@inashivb @victorjulien may I update with the above comment?
Please advice

Reply:
I'm questioning my own logic about the `PASS_IF`. If the "pass" should be last, this should also be true for `PASS_IF`. But what happens if the condition is false? Do we implicitly return `FAIL` in this case? Or would we need something like
```c
PASS_IF(r == 0);
FAIL;
```
But that would violate the "pass should be last rule". We could also do something like `PASS_IF_OR_FAIL(r == 0)` but that becomes a bit too convoluted I think.

So I think I would prefer not introducing a `PASS_IF` after all and just write the tests in such a way that when we reach the end we can issue a `PASS`.

@jasonish any thoughts?

Reply:
PASS_IF is terminal isn't it? It just bundles together a FAIL_IF and a PASS.  I see its only used once. I say we get rid of it. While there is nothing wrong with it, everything else is FAIL type expressions, then we have this one PASS_IF expression which you have to invert your logic for making it hard to read.

Reply:
Please advice on what is to be done further for the issue. Shall i remove the PASS_IF macros and also update #5482 accordingly?

Reply:
Yeah lets not introduce `PASS_IF`. Sorry we put you on the wrong track.